src/drive/web/modules/search/hooks/useSearch.jsx

```diff
@@ -0,0 +1,64 @@
+/* global cozy */
+
+import { useState } from 'react'
+import debounce from 'lodash/debounce'
+import { useClient } from 'cozy-client'
+import { indexFiles } from '../components/helpers'
+import { useMemo } from 'react'
+
+const MAX_SUGGESTIONS = 10
+
+const useSearch = () => {
+  const client = useClient()
+  const [suggestions, setSuggestions] = useState([])
+  const [fuzzy, setFuzzy] = useState(null)
+  const [isBusy, setBusy] = useState(false)
+  const [query, setQuery] = useState('')
+
+  const fetchSuggestions = debounce(value => {
+    onFetchSuggestionsRequested(value)
+  }, 250)
+
+  const onFetchSuggestionsRequested = async value => {
+    setQuery(value)
+    setBusy(true)
+    let currentFuzzy = fuzzy
+    if (currentFuzzy == null) {
+      currentFuzzy = await indexFiles(cozy.client, client)
```

Comment:
Wait what is `cozy.client` here ?

Reply:
That's a good question. `cozy.client` was used in the previous way to search. I didn't take the time to remove it in the PR. Maybe we can do an issue and solve this problem on it's own.